storm/src/main/storm/mesos/MesosNimbus.java

```diff
@@ -577,288 +409,265 @@ public void assignSlots(Topologies topologies, Map<String, Collection<WorkerSlot
       for (WorkerSlot slot : topologyToSlots.getValue()) {
         TopologyDetails details = topologies.getById(topologyId);
         LOG.debug("assignSlots: topologyId: {} worker being assigned to slot: {} with workerCpu: {} workerMem: {}",
-                  topologyId, slot, MesosCommon.topologyWorkerCpu(_conf, details), MesosCommon.topologyWorkerMem(_conf, details));
+                  topologyId, slot, MesosCommon.topologyWorkerCpu(mesosStormConf, details), MesosCommon.topologyWorkerMem(mesosStormConf, details));
       }
     }
 
     synchronized (_offersLock) {
-      computeLaunchList(topologies, slotsForTopologiesNeedingAssignments);
-    }
-  }
+      Map<String, OfferResources> offerResourcesPerNode = MesosCommon.getConsolidatedOfferResourcesPerNode(_offers);
```

Comment:
Maybe mention in a brief comment *why* we need to call this again.
1. because _offers could have changed between `allSlotsAvailableForScheduling()` and `assignSlots()`
2. because we don't store the result of calling `getConsolidatedOfferResourcesPerNode()` in `allSlotsAvailableForScheduling()`.